### PR TITLE
Ningyao finished homework1

### PR DESCRIPTION
**Pull request by AoyamaG**
Please take a look and leave some comments.

### CODE REVIEW

File: course4/homework1/vector2d.h [hunk 1]
```diff
@@ -5,5 +5,23 @@
 struct Vector2d
 {
 	// TODO
+private:
+	double x;
+	double y;
+public:
+	Vector2d() = delete;
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: Why have you decided to do it?

File: course4/homework1/vector2d.h [hunk 2]
```diff
@@ -5,5 +5,23 @@
...
+	Vector2d(double a, double b) : x(a), y(b) {}
+	Vector2d operator+ (const Vector2d& v2) 
+	{
+		return{ x + v2.x, y + v2.y };
+	};
+	bool operator== (const Vector2d& v2)
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: Both t== and + operators should be const

File: course4/homework1/vector2d.h [hunk 3]
```diff
@@ -5,5 +5,23 @@
...
+	{
+		return (x == v2.x) && (y == v2.y) ? true : false;
```
Comment (maciekgajewski) [inline course4/homework1/vector2d.h]: The ? : operation doesn't add anything; it makes no sense to use it.

File: course4/homework1/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+project(homework1)
+include_directories(.)
```
Comment (maciekgajewski) [inline course4/homework1/CMakeLists.txt]: This should never be needed

**Review by maciekgajewski — COMMENTED** [review]
Mostly correct. There is few weird, redundant and unnecessary bits, which should be removed.

File: course4/homework1/vector2d.h (hunk 3 above)
Comment (AoyamaG) [reply course4/homework1/vector2d.h]: Agree. Do not know why I used ?:

File: course4/homework1/vector2d.h (hunk 1 above)
Comment (AoyamaG) [reply course4/homework1/vector2d.h]: I did not want a vector being initialized with random numbers.

File: course4/homework1/CMakeLists.txt (hunk 4 above)
Comment (AoyamaG) [reply course4/homework1/CMakeLists.txt]: I use visual studio. 
project(homework1) creates a solution file for me to open up using vs.
The second line I agree I should delete.

Comment (maciekgajewski) [reply course4/homework1/CMakeLists.txt]: "project" is fine, the "include_directories" is not needed.

**Pull request closed without merging**